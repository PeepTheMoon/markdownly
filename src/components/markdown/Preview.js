import React from 'react';
import styles from 'github-markdown-css/github-markdown.css';
import { useSelector }  from '../../hooks/Context';
import { getMarkdownPreview } from '../../selectors/selectors';

const Preview = () => {
  const __html = useSelector(getMarkdownPreview);

  return (
    <>
      <div className={styles['markdown-body']} dangerouslySetInnerHTML={{ __html }}></div>
    </>
  );
};

export default Preview;
