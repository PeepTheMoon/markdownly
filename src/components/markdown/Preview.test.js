import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview';

jest.mock('../../hooks/Context');

describe('Preview component', () => {
  it('renders preview', () => {
    const wrapper = shallow(<Preview />);
    expect(wrapper).toMatchSnapshot();
  });
});
