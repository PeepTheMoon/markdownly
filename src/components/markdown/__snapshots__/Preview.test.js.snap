// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview component renders preview 1`] = `
<Fragment>
  <div
    className="markdown-body"
    dangerouslySetInnerHTML={
      Object {
        "__html": undefined,
      }
    }
  />
</Fragment>
`;
